### PR TITLE
[oneDNN] Tensor copy fix to oneDNN tensors

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Bug fixes

### PR changes
Others

### Describe
TensorCopy was not working properly for oneDNN tensors which may result in some undefined behaviour.

### CODE REVIEW

**Comment by jczaja** [comment]
@arlesniak  Please review

File: paddle/fluid/framework/tensor_util.cc [hunk 1]
```diff
@@ -43,20 +43,27 @@ void TensorCopy(const Tensor& src, const platform::Place& dst_place,
...
+  auto src_ptr = src.data<void>();
 #ifdef PADDLE_WITH_MKLDNN
   dst->set_format(src.format());
 #endif
-  auto src_place = src.place();
-  auto src_ptr = src.data<void>();
-  auto dst_ptr = dst->mutable_data(dst_place, src.type());
+  // oneDNN tensors due to padding may be of bigger size
+  // than numel()*size(type())
+  auto dst_ptr =
```
Comment (arlesniak) [inline paddle/fluid/framework/tensor_util.cc:52]: Maybe the new block of code should be guarded with: `#ifdef PADDLE_WITH_MKLDNN` ?

File: paddle/fluid/framework/tensor_util.cc [hunk 2]
```diff
@@ -43,20 +43,27 @@ void TensorCopy(const Tensor& src, const platform::Place& dst_place,
...
+          : dst->mutable_data(dst_place, src.type());
 
   if (src_ptr == dst_ptr && src_place == dst_place) {
     VLOG(3) << "Skip copy the same data async from " << src_place << " to "
             << dst_place;
     return;
   }
 
-  auto size = src.numel() * SizeOfType(src.type());
+  auto size = src.layout() == DataLayout::kMKLDNN
```
Comment (arlesniak) [inline paddle/fluid/framework/tensor_util.cc:66]: ditto

File: paddle/fluid/framework/tensor_util.cc (hunk 1 above)
Comment (jczaja) [reply paddle/fluid/framework/tensor_util.cc:52]: ok

File: paddle/fluid/framework/tensor_util.cc (hunk 2 above)
Comment (jczaja) [reply paddle/fluid/framework/tensor_util.cc:66]: done

**Comment by jczaja** [comment]
@arlesniak  please continue your review

**Review by arlesniak — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 This PR is changing common code (tensor_utils.cc) it would be great if anyone experienced in that area took a look at that PR.

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**